File: mall-manager-web/src/main/webapp/js/service/specificationService.js
/**
 * 规格服务层
 * Created by Wwl on 2019/3/14
 */
app.service('specificationService',function($http){

    //增加
    this.add = (entity) => $http.post('../specification/add.do',entity);

    //修改
    this.update = (entity) => $http.post('../specification/update.do',entity);

    //删除
    this.delete = (ids) => $http.get('../specification/delete.do?ids='+ids);

    //查询实体
    this.findOne = (id) => $http.get('../specification/findOne.do?id='+id);

    //查询列表
    this.findList = () => $http.get('../specification/list.do');

    //分页查询
    this.findPage = (page, size) => $http.get('../specification/listByPage.do?page=' + page + '&size=' + size);

    //搜索
    this.search = (page,size,searchEntity) => $http.post('../specification/search.do?page='+page+'&size='+size,searchEntity);

    //下拉列表数据
    this.findSpecList = () => $http.get('../specification/findOptionList.do');

});

File: mall-goods-interface/src/main/java/com/mall/goods/service/SpecificationService.java
package com.mall.goods.service;

import java.util.List;
import java.util.Map;

import com.mall.pojo.Specification;

import com.mall.pojogroup.SpecificationGroup;
import common.pojo.PageResult;

/**
 * 规格服务接口
 * @author Wwl
 *
 */
public interface SpecificationService {

	/**
	 * 增加
	 * @param specificationGroup
	 */
	public void add(SpecificationGroup specificationGroup);

	/**
	 * 修改
	 * @param specificationGroup
	 */
	public void update(SpecificationGroup specificationGroup);

	/**
	 * 根据Id获取实体
	 * @param id
	 * */
	public SpecificationGroup findOne(Long id);

	/**
	 * 查询所有
	 * @return
	 */
	public List<Specification> findAll();

	/**
	 * 查询所有-分页
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);

	/**
	 * 根据条件查询，分页
	 * @param specification
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Specification specification,int pageNum,int pageSize);

	/**
	 * 删除
	 * @return
	 */
	public void delete(Long[] ids);

	/**
	 * 下拉列表
	 * @return
	 */
	public List<Map> findOptionList();
}
